app/src/main/java/com/elkcreek/rodneytressler/intermediatepokedex/ui/PokedexView/PokedexPresenter.java

```diff
@@ -53,39 +49,48 @@ public void searchClicked(String pokemonName) {
         view.showProgressBar();
         view.clearPokemonInput();
 
-        checkDatabase(pokemonName);
+        findPokemon(pokemonName);
+    }
+
+    private Observable<PokemonApi.Pokemon> findInDatabase(String name) {
```

Comment:
Wrapper around pokemonDatabaseService that turns the single into an observable to make it more cohesive